### PR TITLE
Resolve warnings in Apple-clang

### PR DESCRIPTION
**Pull request by psakievich**
## Description of the pull-request

I got tired of the seeing all the warnings in about deleted constructors. This compiles with `-Werror` on my Mac at least for the base code and OpenFAST turned on.  I'll do a follow up PR with other TPL's turned on.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [x] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [ ] Linux
    - [x] MacOS
  - Compilers 
    - [ ] GCC
    - [x] LLVM/Clang
    - [ ] Intel compilers
    - [x] NVIDIA CUDA
- [x] Compiles without warnings
- [x] Passes all unit tests
- [ ] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [ ] New regression tests exercising the new code

### CODE REVIEW

File: src/NaluParsing.C [hunk 1]
```diff
@@ -167,8 +167,6 @@ namespace sierra
         }
       }
 
-      const YAML::Node& oversetUserData = node["overset_user_data"];
```
Comment (sayerhs) [inline src/NaluParsing.C:170]: This breaks builds with TIOGA enabled

**Review by sayerhs — CHANGES_REQUESTED** [review]
Breaks build with TIOGA overset enabled.

**Comment by psakievich** [comment]
Anyone know how `KOKKOS_FUNCTION constuctor() =delete;` will behave?  I'll test it out.  I'm guessing I should probably just remove the kokkos macros too though.

File: src/NaluParsing.C (hunk 1 above)
Comment (psakievich) [reply src/NaluParsing.C:170]: Ah oh yeah.  Okay I'll move this inside the `#ifdef`

Comment (psakievich) [reply src/NaluParsing.C:170]: K pushed a fix for this.  I've added the WIP tag to prevent merging until I finish testing the cuda build.

**Comment by alanw0** [comment]
@psakievich I know there is a KOKKOS_DEFAULTED_FUNCTION, perhaps there is a KOKKOS_DELETED_FUNCTION but I'm not sure. If the function is deleted then it doesn't seem like it should need to be decorated...

**Comment by psakievich** [comment]
Well it looks like the cuda build doesn't care if you leave the macro there, but I think I will remove them all anyways because I think it's a better practice.

**Review by sayerhs — APPROVED** [review]

**Pull request merged**